packages/gatsby/src/commands/build.ts

```diff
@@ -134,7 +137,12 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
     rewriteActivityTimer.end()
   }
 
-  await processPageQueries()
+  await runPageQueries({
+    queryIds,
+    graphqlRunner,
+    parentSpan: buildSpan,
+    store,
+  })
```

Comment:
There is no need to leak staticQueryIds into pageQueries
```suggestion
  await runPageQueries({
    queryIds: queryIds.pageQueryIds,
    graphqlRunner,
    parentSpan: buildSpan,
    store,
  })
```

Reply:
I can separate them, but I can't do it like this. They can't share a prop name, because the machine will pass the same context to each of them. I'd need to change it to add separate pageQueryIds and staticQueryIds props

Reply:
My gut feeling says something is wrong with the state machine.

Reply:
I do it like this because we group the query ids once in calculateDirtyQueries, and then pass them in context to the two query running states. I'm happy to move them into the top level of the context, but didn't feel it made much difference, as they're never set independently.

Reply:
Please note that we don't actually need to even group queries into page/static buckets anymore - we chatted about this yesterday with @ascorbic and reason for the split was writing webpack compilation hash into `page-data` files (in first iteration of getting rid of "pages manifest" way back).

We later moved webpack compilation hash to app-data, but the split remained (mostly because it wasn't really causing problems, so there was not much need to clean it up).

We chatted about getting rid of grouping later down the road - just it was not in scope of this PR as this one meant only to do 1:1 refactor to prepare to move to state machine world without functionally changing anything